### DISCUSSION TITLE
How to get the OpenGL texture handle from a filament texture?

### DISCUSSION THREAD

**Discussion by yavuz650** [google/filament#6401] [Q&A] [answered]
I'm doing off-screen rendering and I need to access the textures that filament renders into. I looked at the Texture interface, but there is no way to obtain the texture handle from there.

Is there a way to get the texture handles?

**Comment by bejado** [comment, marked as answer]
Depending on what you're trying to do, you could go the other route: "import" an OpenGL texture into Filament. See the [import](https://github.com/google/filament/blob/main/filament/include/filament/Texture.h#L241) method.

**Reply by yavuz650** [reply]
This could work. I'll try and see what happens. Thank you.

**Comment by pixelflinger** [comment]
There is no way to get the OpenGL handle from `Texture`. The only solution is to import a texture instead.
